### PR TITLE
fix: Handle payment errors during user flow

### PR DESCRIPTION
**Pull request by zachaysan**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

[Ticket reference](https://github.com/orgs/Flagsmith/projects/2/views/11?pane=issue&itemId=35443590)

Before this PR we were treating all errors from the chargebee client as one thing and we responded to a user that was in the process of signing up by issuing a 500 error, we were also logging all the chargebee errors as exceptions even though a declined credit card shouldn't be. Now, if the error is payment related, a `400` error gets sent to the client with a message to contact their administrator. 

## How did you test this code?

A combination of manual probing through the code with the test suite on and two new tests. 

## Remaining TODOs
- [x] Figure out what other callsites there are that will be impacted by this change, if any.
- [x] See if there are any other other areas of the codebase with invite-related code need to handle payment failures.
- [x] Manually test this somehow. I have access to flagsmith-test.chargebee.com but I'm not sure what server it is connected to. Is that staging?
- ~[ ] Team up with @novakzaballa and get the frontend hooked up so the change will actually matter.~

### CODE REVIEW

File: api/organisations/chargebee/chargebee.py [hunk 1]
```diff
@@ -177,6 +186,14 @@ def add_single_seat(subscription_id: str):
         )
 
     except ChargebeeAPIError as e:
+        api_error_code = e.json_obj["api_error_code"]
+        if api_error_code in CHARGEBEE_PAYMENT_ERROR_CODES:
+            logger.warning(
+                "Payment declined during additional seat upgrade to a "
+                f"CB subscription for subscription_id {subscription_id}"
+            )
```
Comment (zachaysan) [inline api/organisations/chargebee/chargebee.py]: I wasn't sure whether we wanted to log the specific payment failure here or not.

Comment (matthewelwell) [reply api/organisations/chargebee/chargebee.py]: I think probably best to include it, yep.

Comment (matthewelwell) [reply api/organisations/chargebee/chargebee.py]: I have added this so we can get this merged 👍

**Review by matthewelwell — APPROVED** [review]

File: api/organisations/chargebee/chargebee.py (hunk 1 above)
Comment (zachaysan) [reply api/organisations/chargebee/chargebee.py]: Thank you!!

**Pull request merged**